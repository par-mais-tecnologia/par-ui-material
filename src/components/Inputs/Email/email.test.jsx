import React from "react";
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EmailInput from './email'
configure({ adapter: new Adapter() });

export const test = describe('E-mail', () => {
    it('shallow E-mail', () => {
        const wrapper = shallow(
            <EmailInput/>
        );
        expect(wrapper).toBeTruthy();
    });

    it('shallow E-mail match Snapshot', () => {
        const wrapper = shallow(
            <EmailInput/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});