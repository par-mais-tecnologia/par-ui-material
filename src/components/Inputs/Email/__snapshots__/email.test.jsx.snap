// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E-mail shallow E-mail match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailInput
    errorMessage=""
    hasError={false}
    label="E-mail"
    required={false}
    showError={true}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-describedby": "email-form-control",
      "children": Array [
        <TextField
          id="email-input"
          label="E-mail"
          margin="normal"
          onBlur={[Function]}
          required={false}
          select={false}
          value=""
        />,
        <WithStyles(FormHelperText)
          id="email-form-control"
        >
           
           
        </WithStyles(FormHelperText)>,
      ],
      "required": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "error": undefined,
          "id": "email-input",
          "label": "E-mail",
          "margin": "normal",
          "onBlur": [Function],
          "onChange": undefined,
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
          "id": "email-form-control",
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "email-form-control",
        "children": Array [
          <TextField
            id="email-input"
            label="E-mail"
            margin="normal"
            onBlur={[Function]}
            required={false}
            select={false}
            value=""
          />,
          <WithStyles(FormHelperText)
            id="email-form-control"
          >
             
             
          </WithStyles(FormHelperText)>,
        ],
        "required": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "error": undefined,
            "id": "email-input",
            "label": "E-mail",
            "margin": "normal",
            "onBlur": [Function],
            "onChange": undefined,
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
            "id": "email-form-control",
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
